### PR TITLE
Use proper empty values for job info

### PR DESCRIPTION
**Pull request by davepagurek**
@andrew749 wants null working directories and empty objects instead of empty strings

### CODE REVIEW

File: app/models/job_type.rb [hunk 1]
```diff
@@ -3,4 +3,13 @@
 class JobType < ApplicationRecord
   belongs_to :user
   has_many :jobs
+
+  before_save :nullify_blanks
+
+  private
+
+  def nullify_blanks
```
Comment (AndrewMcBurney) [inline app/models/job_type.rb]: I would change this to 
```ruby
def nullify_blanks
  self.environment_variables ||= "{}"
  self.working_directory ||= nil
end
```

File: app/models/job_type.rb [hunk 2]
```diff
@@ -3,4 +3,13 @@
 class JobType < ApplicationRecord
   belongs_to :user
   has_many :jobs
+
+  before_save :nullify_blanks
```
Comment (AndrewMcBurney) [inline app/models/job_type.rb]: Can you please move this above the associations to be consistent with the other models?

File: app/models/job_type.rb (hunk 1 above)
Comment (davepagurek) [reply app/models/job_type.rb]: Does a blank string evaluate to false in a boolean context though?

Comment (davepagurek) [reply app/models/job_type.rb]:
```
18:44 $ irb
2.4.1 :001 > !!""
 => true
```

Comment (AndrewMcBurney) [reply app/models/job_type.rb]: Ahh, good call.

**Pull request merged**